src/query/ast/src/parser/statement.rs

```diff
@@ -2150,13 +2150,11 @@ pub fn user_option(i: Input) -> IResult<UserOptionItem> {
 pub fn user_identity(i: Input) -> IResult<UserIdentity> {
     map(
         rule! {
-            #parameter_to_string ~ ( "@" ~ #literal_string )?
```

Comment:
how about `#parameter_to_string ~ ( "@" ~  "'%'" )?` ? 

so this may compatbile with the existed `'user'@'%'` queries, but do not allow any other host values.